theme/static/js/resourceLandingAjax.js

```diff
@@ -893,6 +910,7 @@ function filetype_keywords_update_ajax_submit() {
                         li.find('span').text(resKeywords[i]);
                         li.append('&nbsp;<a><span class="glyphicon glyphicon-remove-circle icon-remove"></span></a>')
                         $("#lst-tags").append(li);
+                        $("#lst-tags").find(".icon-remove").click(onRemoveKeyword);
```

Comment:
@pkdash, same here. https://stackoverflow.com/a/34896387

Reply:
Would it work even if we are adding a new ```span``` element here as shown in line#911?

Reply:
@pkdash, yes. Basically you use a component that is available during the $(document).ready method as reference and bind from there. Here is an example from our code:

```
$("#extraMetaTable").on("click", ".btn-edit-extra-metadata", function () {
        var index = $(this).attr("data-loop-counter");
        showAddEditExtraMetaPopup(true, index);
    });
```
In this case, `#extraMetaTable` is the referenced element that is not expected to change. From there we bind a click event to all `.btn-edit-extra-metadata`. This will keep the binding updated even for dynamically created elements.

Reply:
Tried to move the line#913 into document.ready() of composite-resource.js. But then the delete of keywords that are added to the resource as a result of adding keywords to file level does not work. So not making the change.